internal/databases/postgres/daemon_handler.go

```
@@ -0,0 +1,176 @@
+	}
+	_, err = h.fd.Write([]byte{'O'})
+	if err != nil {
+		tracelog.ErrorLogger.Printf("failed to write in socket: %v\n", err)
+		return err
+	}
+	return nil
+}
+
+func messageValidation(messageBody []byte) error {
+	if len(messageBody) < 24 {
+		if len(messageBody) > 0 {
+			tracelog.ErrorLogger.Println("incorrect message accepted")
+			return fmt.Errorf("incorrect message accepted: %s", string(messageBody))
+		}
+		return fmt.Errorf("empty message accepted")
+	}
+	return nil
+}
+
+func GetMessageHandler(messageType byte, c net.Conn, uploader *WalUploader) SocketMessageHandler {
+	switch messageType {
+	case 'C':
+		return &CheckMessageHandler{CheckType, c, uploader}
+	case 'F':
+		return &ArchiveMessageHandler{FileNameType, c, uploader}
+	default:
+		return nil
+	}
+}
+
+type SocketMessageReader struct {
+	c net.Conn
+}
+
+func GetMessageReader(c net.Conn) *SocketMessageReader {
+	return &SocketMessageReader{c}
+}
+
+// Next method reads messages sequentially from the Reader
+func (r SocketMessageReader) Next() (messageType byte, messageBody []byte, err error) {
+	messageParameters := make([]byte, 3)
+	_, err = io.ReadFull(r.c, messageParameters)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to read from socket, err: %v \n", err)
+		return 'E', nil, err
+	}
+	messageType = messageParameters[0]
+	var messageLength uint16
+	l := bytes.NewReader(messageParameters[1:3])
+	err = binary.Read(l, binary.BigEndian, &messageLength)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to read message length, err: %v \n", err)
+		return 'E', nil, err
+	}
+	messageBody = make([]byte, messageLength-3)
+	_, err = io.ReadFull(r.c, messageBody)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to read from socket, err: %v \n", err)
+		return 'E', nil, err
+	}
+	return messageType, messageBody, err
+}
+
+// HandleDaemon is invoked to perform daemon mode
+func HandleDaemon(uploader *WalUploader, pathToSocket string) {
+	_ = os.Remove(pathToSocket)
```

do not ignore the errors